### PR TITLE
Link from parser to an initial solution

### LINKED ISSUE #4: Data structure
When #3 will be merge `parser()` will extract raw data. We need to read that data and generate the data structure chosen in [An efficient implementation of a VNS/ILS heuristic for a real-life car sequencing problem](https://doi.org/10.1016/j.ejor.2007.02.003). Code should be write directly in `parser.jl`

Blocked by #2

### PR DESCRIPTION
**Pull request by Jonathan-Fontaine44**
Closes #4

### CODE REVIEW

File: src/solution.jl [hunk 1]
```diff
@@ -13,10 +14,21 @@ mutable struct Solution
...
     )
 end
+
+# Build an initial
+function init_solution(nom_fichier::String, type_fichier::String)
+    # Read in data files
+    instance = parser(nom_fichier, type_fichier)
+
+    n = length(instance.color_code)
+    m = length(instance.HPRC) + length(instance.LPRC) # number of ratio
```
Comment (k-fretaud) [inline src/solution.jl]: Why not  `m = instance.nb_HPRC + instance.nb_LPRC` ?

File: src/solution.jl [hunk 2]
```diff
@@ -13,10 +14,21 @@ mutable struct Solution
         # M3_ij is the number of subsequences in which the number of
         # cars that require oj is greater than or equal to p(oj)
 
-    Solution(nC,nO) = new(
-        zeros(Int,nC),
+    Solution(nC::Int,nO::Int) = new(
+        collect(1:nC),
```
Comment (k-fretaud) [inline src/solution.jl:18]: I think `collect(1:nC)` is too much for an empty solution. Initialization with `zeros` could be an efficient "default value" when we will implement a greedy algorithm.

**Review by k-fretaud — COMMENTED** [review]
Two points discussed, looks good otherwise.

File: src/solution.jl (hunk 1 above)
Comment (Jonathan-Fontaine44) [reply src/solution.jl]: In fact it would be better, I will change that.

File: src/solution.jl (hunk 2 above)
Comment (Jonathan-Fontaine44) [reply src/solution.jl:18]: I think a default value should be a solution, and zeros is not. However I can accept to change if other people agree with you.

**Review by k-fretaud — APPROVED** [review]

File: src/solution.jl (hunk 2 above)
Comment (remi-garcia) [reply src/solution.jl:18]: I agree with @Jonathan-Fontaine44, zeros is not a good solution. For an empty solution we could go for `nothing` as well.

**Review by BenoitLoger — APPROVED** [review]

**Comment by remi-garcia** [comment]
This shouldn't have been merged imho. M1, M2 and M3 are not consistent with sequence. We shouldn't do that.

**Comment by Jonathan-Fontaine44** [comment]
Then I saw three options : 

1. Compute M1, M2 and M3 Matrix for the initial solution.
2. Creating an empty solution full of zeros and undef.
3. Delete the function `init_solution` and the constructor, and let the greedy algorithm be the initial solution as it should be.

**Pull request merged**